File: server/python/app/infrastructure/mappers/payment_details/payment_references_mapper.py
from typing import Optional
from app.application.domain.payments.payment_details.payment_references import PaymentReferences
from onlinepayments.sdk.domain.payment_references import PaymentReferences as PaymentReferencesSdk

class PaymentReferencesMapper:

    @staticmethod
    def map_from_sdk_response(response: Optional[PaymentReferencesSdk]) -> Optional[PaymentReferences]:
        if response is None:
            return None

        return PaymentReferences(
            merchant_reference = response.merchant_reference,
            merchant_parameters = response.merchant_parameters
        )

File: server/ruby/app/infrastructure/mappers/payment_details/payment_references_mapper.rb
module Infrastructure
  module Mappers
    module PaymentDetails
      class PaymentReferencesMapper
        def self.from_sdk_response(response)
          return nil if response.nil?

          Business::Domain::Payments::PaymentDetails::PaymentReferences.new.tap do |dto|
            dto.merchant_parameters = response.merchant_parameters
            dto.merchant_reference = response.merchant_reference
          end
        end
      end
    end
  end
end